client/components/login.js

```diff
@@ -1,33 +1,98 @@
 import React, {Component} from "react";
 import validator from "validator";
 import classNames from "classnames";
-import update from 'immutability-helper';;
+import update from 'immutability-helper';
 
 export class LogInForm extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            user: {}
+            user: {},
+            validUsername: null,
+            validPassword: null
         };
     }
 
-    handleSubmit(e) {
+    handleSubmit = (e) => {
         e.preventDefault();
-        console.log("handled submit");
-    }
+        if (this.state.validUsername && this.state.validPassword) {
+            this.props.handleLogIn(this.state.user);
+        }
+    };
+
+    handleUsernameChange = (e) => {
+        const val = e.target.value;
+        if (!validator.isEmpty(val) && validator.isAscii(val)) {
+            this.setState({
+                validUsername: true,
+                user: update(this.state.user, {username: {$set: val}})
```

Comment:
why are you using update?

Object.assign({}, this.state.user, {username: val});